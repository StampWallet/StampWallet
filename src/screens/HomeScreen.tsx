import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

import { useFonts } from 'expo-font';
import { Roboto_400Regular, Roboto_700Bold_Italic } from '@expo-google-fonts/roboto';

import CustomButton from '../components/CustomButton';

export default function HomeScreen({ navigation }) {
  const [fontsLoaded] = useFonts({
    'Roboto-Regular': Roboto_700Bold_Italic,
  });

  if (!fontsLoaded) {
    return null;
  }

  return (
    <View style={styles.container}>
      <CustomButton
        onPress={() => {
          navigation.push('LogInScreen');
        }}
        title='Sign in'
      />
      <Text style={{ fontFamily: 'Roboto-Regular' }}>asdasdasdas</Text>
      <CustomButton
        onPress={() => {
          navigation.push('RegistrationScreen');
        }}
        title='Register'
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    backgroundColor: '#FAF9F6',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  textInput: {
    height: 60,
    margin: 12,
    padding: 10,
    fontSize: 18,
  },
});
