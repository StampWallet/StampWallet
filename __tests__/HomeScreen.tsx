import React from 'react';
import { describe, expect } from '@jest/globals';
import { render, fireEvent } from 'react-native-testing-library';
import '@testing-library/jest-native/extend-expect';

import HomeScreen from '../screens/HomeScreen';

describe('Home screen tests', () => {
  it('checks if rendered correctly', () => {
    const tree = render(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('checks if proper register form screen called', () => {
    const chooseFormScreen = jest.fn();
    const tree = render(<HomeScreen />)

    const registerButton = tree.getByText('Register');
    expect(registerButton).toBeVisible();
    fireEvent.press(registerButton);
    expect(chooseFormScreen).toBeCalledWith('REGISTER');

    const loginButton = tree.getByText('Sign in');
    expect(loginButton).toBeVisible();
    fireEvent.press(loginButton);
    expect(chooseFormScreen).toBeCalledWith('LOGIN');

  })
});
