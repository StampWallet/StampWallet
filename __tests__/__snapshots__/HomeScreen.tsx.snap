// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen tests checks if rendered correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    This is home screen
  </Text>
</View>
`;
